### PR TITLE
V2 Drawer (Top, Bottom, Left, Right) implementation & Demo app intergration

### PR DESCRIPTION
**Pull request by mishramayank1**

### CODE REVIEW

File: fluentui_core/build.gradle [hunk 1]
```diff
@@ -77,6 +78,7 @@ dependencies {
     implementation "androidx.compose.ui:ui-text:$composeVersion"
     implementation "androidx.compose.ui:ui-graphics:$composeVersion"
     implementation "androidx.compose.ui:ui-unit:$composeVersion"
+    implementation "androidx.compose.ui:ui-util:$composeVersion"
```
Comment (NamanPandey) [inline fluentui_core/build.gradle:81]: Shouldn't we include all compose.ui libraries? Including one lib at a time isn't the most optimised way.

File: fluentui_drawer/src/main/java/com/microsoft/fluentui/tokenized/drawer/Drawer.kt [hunk 2]
```diff
@@ -0,0 +1,753 @@
...
+}
+
+/**
+ * State of the [Drawer] composable.
+ *
+ * @param initialValue The initial value of the state.
+ * @param confirmStateChange Optional callback invoked to confirm or veto a pending state change.
+ */
+@Suppress("NotCloseable")
+@Stable
```
Comment (NamanPandey) [inline fluentui_drawer/src/main/java/com/microsoft/fluentui/tokenized/drawer/Drawer.kt:79]: pls check if this is required.

File: fluentui_drawer/src/main/java/com/microsoft/fluentui/tokenized/drawer/Drawer.kt [hunk 3]
```diff
@@ -0,0 +1,753 @@
...
+            swipeableState.snapTo(targetValue)
+
+    /**
+     * The target value of the drawer state.
+     *
+     * If a swipe is in progress, this is the value that the Drawer would animate to if the
+     * swipe finishes. If an animation is running, this is the target value of that animation.
+     * Finally, if no swipe or animation is in progress, this is the same as the [currentValue].
+     */
+    @Suppress("OPT_IN_MARKER_ON_WRONG_TARGET")
```
Comment (NamanPandey) [inline fluentui_drawer/src/main/java/com/microsoft/fluentui/tokenized/drawer/Drawer.kt:175]: why is this required?

**Review by NamanPandey — APPROVED** [review]

File: fluentui_drawer/src/main/java/com/microsoft/fluentui/tokenized/drawer/Drawer.kt (hunk 3 above)
Comment (mishramayank1) [reply fluentui_drawer/src/main/java/com/microsoft/fluentui/tokenized/drawer/Drawer.kt]: removed

File: fluentui_drawer/src/main/java/com/microsoft/fluentui/tokenized/drawer/Drawer.kt (hunk 2 above)
Comment (mishramayank1) [reply fluentui_drawer/src/main/java/com/microsoft/fluentui/tokenized/drawer/Drawer.kt]: removed

File: fluentui_drawer/src/main/java/com/microsoft/fluentui/tokenized/drawer/Drawer.kt [hunk 4]
```diff
@@ -0,0 +1,780 @@
...
+        }
+
+        val fullWidth = modalDrawerConstraints.maxWidth.toFloat()
+        var drawerWidth by remember(fullWidth) { mutableStateOf(fullWidth) }
+        //Hack to get exact drawerHeight wrt to content.
+        val visible = remember { mutableStateOf(true) }
+        if (visible.value) {
+            Box(modifier = Modifier.onPlaced { layoutCoordinates ->
+                drawerWidth = layoutCoordinates.size.width.toFloat()
+                visible.value = false
```
[the remark this answers is not in the document]
Comment (mishramayank1) [reply fluentui_drawer/src/main/java/com/microsoft/fluentui/tokenized/drawer/Drawer.kt]: ".value" used to explicitly mention that it invokes recomposition on changing values & take further action.
